src/Pages/Main/MainSearch.js

```diff
@@ -0,0 +1,197 @@
+import { Link, withRouter } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { searchBtnSvg } from "../../config";
+import styled from "styled-components";
+
+const MainSearch = (props) => {
+  const [searchInput, setSearchInput] = useState("");
+  const [searchDatas, setSearchDatas] = useState([]);
+  const [searchResult, setSearchResult] = useState([]);
+  const [stateSearchModal, setStateSearchModal] = useState(false);
+
+  const handleInputChange = (e) => {
+    const {
+      target: { value },
+      keyCode,
+    } = e;
+
+    const resultFunction = searchDatas
+      .filter((data) => {
+        return data.cardUserId.toLowerCase().includes(value.toLowerCase());
+      })
+      .slice(0, 5);
+    console.log(resultFunction);
+    value.length && keyCode === 13 && props.history.push("/search/${value}");
+    value.length ? setSearchResult(resultFunction) : setSearchResult([]);
+    value.length ? setStateSearchModal(true) : setStateSearchModal(false);
```

Comment:
```suggestion
    setSearchResult(value.length ? resultFunction : [])
    setStateSearchModal(!!value.length)
```
함수보다는 들어갈 값에 대해 삼항연산자를 활용하면 좋겠네요

Reply:
수정했습니다 항상 감사합니다 종택님! 🙌